polaris-react/src/components/AlphaFilters/AlphaFilters.tsx

```diff
@@ -298,7 +298,9 @@ export function AlphaFilters({
         <FilterPill
           key={filterKey}
           {...pinnedFilter}
-          initialActive={hasMounted.current && !pinnedFilter.pinned}
+          initialActive={
+            hasMounted.current && !pinnedFilter.pinned && !appliedFilter
```

Comment:
LGTM. An `appliedFilter` should never automatically be open. Do you mind adding a test and running `yarn changeset`?